src/main.ts

```diff
@@ -770,13 +772,15 @@ function generateOneofProperty(
     fields.map((f) => {
       let fieldName = maybeSnakeToCamel(f.name, options);
       let typeName = toTypeName(ctx, messageDesc, f);
-      return code`{ $case: '${fieldName}', ${fieldName}: ${typeName} }`;
+      return code`{ ${ctx.options.useReadonlyTypes ? "readonly " : ""}$case: '${fieldName}', ${
+        ctx.options.useReadonlyTypes ? "readonly " : ""
+      }${fieldName}: ${typeName} }`;
     }),
     { on: " | " }
   );
 
   const name = maybeSnakeToCamel(messageDesc.oneofDecl[oneofIndex].name, options);
-  return code`${name}?: ${unionType},`;
+  return code`${ctx.options.useReadonlyTypes ? "readonly " : ""}${name}?: ${unionType},`;
```

Comment:
@jonaskello wdyt of a helper method like:

```
code`${maybeReadonly(ctx)}${name}...`

function maybeReadonly(ctx: ...): string {
  return useReadonlyTypes ? "readonly " : "";
}
```

If you'd like to try that, that'd be cool, but otherwise lgtm! Feel free to merge and release (just added you as a collaborator so I believe you should be able to do that; lmk if not).

Reply:
Yes, good idea that would make the code easier to read. Replaced many ternary expressions with two helpers,  ´maybeReadonly()` and `maybeAsAny()`-